tests/planner.spec.ts

```diff
@@ -0,0 +1,142 @@
+import { test, expect, Page } from "@playwright/test";
+
+const testForArcanistPlanning = (
+  ...arcanists: {
+    name: string
+    insight: [number, number]
+    level: [number, number]
+    resonance: [number, number]
+    expectMaterials: {
+      name: string
+      count: string
+    }[]
+  }[]
+) => {
+  return async ({ page }: { page: Page }) => {
+    const setupStatus = async (blockName: string, status: [string, string]) => {
+      if (status.length !== 2) { return }
+
+      const settingBlock = page.locator('.edit-overlay')
+        .locator(`:below(:text("${blockName}"))`)
+        .first()
+
+      for (let i = 0; i < status.length; i++) {
+        await settingBlock.getByRole('button').nth(i).click()
+        const fromBlock = settingBlock.locator('div:below(button)')
+          .filter({ has: page.getByRole('button') })
+          .first()
+        await fromBlock.getByRole('button', { name: status[i], exact: true }).click()
+      }
+    }
+
+    for (let i = 0; i < arcanists.length; i++) {
+      const arcanist = arcanists[i]
+      await page.getByRole("button", { name: "Add Arcanist" }).click()
+      await page.getByLabel('Show Unreleased Arcanists').check()
+      await page.getByText(arcanist.name).click()
+
+      await setupStatus('CURRENT LEVEL', [
+        arcanist.insight[0] ? `Option ${arcanist.insight[0]}` : '', `${arcanist.level[0]}`
+      ])
+      await setupStatus('GOAL LEVEL', [
+        arcanist.insight[1] ? `Option ${arcanist.insight[1]}` : '', `${arcanist.level[1]}`
+      ])
+      await setupStatus('RESONANCE', arcanist.resonance.map(item => `${item}`) as [string, string])
+
+
+      const materialsBlock = page.locator('div:below(button:text("Save"))')
+        .filter({ has: page.getByRole('img') })
+        .first()
+      await Promise.all(
+        arcanist.expectMaterials.map(material => expect(
+          materialsBlock.locator(`[data-tip="${material.name}"]`).getByText(material.count)
+        ).toBeVisible())
+      )
+
+      await page.getByRole('button', { name: 'Save' }).click()
+    }
+
+    await expect(page).toHaveScreenshot();
+  }
+}
+
+// test.describe("Arcanist planning", () => {
+test.beforeEach(({ page }) => {
+  page.goto("/Kornblume/planner")
+})
+
+test(
+  "Should correctly plan several arcanists",
+  testForArcanistPlanning(
```

Comment:
this interface is very clear
btw, I think tracker.spec could also have a similar refactor :3

Reply:
Which interface?
tracker.spec has the same feature doesn't it? I imitated your page XD

Reply:
uh, I just think this interface seems clearer